src/token/api/src/canister.rs

```diff
@@ -51,8 +52,15 @@ pub enum CanisterUpdate {
     MinCycles(u64),
 }
 
-#[allow(non_snake_case)]
-pub trait TokenCanisterAPI: Canister + Sized + Auction {
+#[cfg(feature = "auction")]
+pub trait TokenCanisterTrait: Canister + Sized + Auction {}
```

Comment:
This works for now but this will fall apart very quickly if we ever need to do this with another feature.

Reply:
Agreed. What will happen once we have 3+ features on that?

Reply:
In any case in this example there could be a separate trait that will implement the necessary supertrait as needed. e.g.
```rust
#[cfg(feature = "a")]
pub trait A: FeaturedA

#[cfg(not(feature = "a"))]
pub trait A

pub trait TokenCanisterTrait: Canister + Sized + A
```